cinema/views.py

```diff
@@ -97,12 +100,27 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
         if self.action == "retrieve":
             return MovieDetailSerializer
+        if self.action == "upload_image":
+            return MovieImageSerializer
 
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        url_path="upload-image"
+    )
+    def upload_image(self, request, pk=None):
+        movie = get_object_or_404(Movie, pk=pk)
+        serializer = self.get_serializer(movie, data=request.data)
+        print(request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
you can simplify it calling serializer.is_valid(raise_exception=True) separately, thus there won't be a need to add if statement